src/stactools/core/utils/raster_footprint.py

```diff
@@ -0,0 +1,301 @@
+    If the initial low-vertex polygon in the native CRS is not densified, this can result
+    in a reprojected polygon that does not accurately represent the data footprint. For
+    example, a MODIS asset represented by a rectangular 5 point Polygon in a sinusoidal
+    projection will reproject to a parallelogram in EPSG 4326, when it would be more
+    accurately represented by a polygon with two parallel sides and two curved sides.
+    The difference between these representations is even greater the further away from
+    the meridian and equator the asset is located.
+
+    After reprojection to EPSG 4326, the footprint may have more points than desired.
+    This can be simplified to a polygon with fewer points that maintain a minimum distance
+    between each point.
+
+    Args:
+        item (Item): The PySTAC Item to extend.
+        asset_names (Optional[List[str]]): The names of the assets for which to extract footprints.
+        scale (int): A factor to scale out the shape to. Defaults to 1.
+        precision (int): The number of decimal places to include in the coordinates for the
+            reprojected geometry. Defaults to 3 decimal places.
+        densification_factor (Optional[int]): The factor by which to increase point density within
+            the polygon.
+            A factor of 2 would double the density of points (placing one new point between each
+            existing pair
+            of points), a factor of 3 would place two points between each point, etc.
+        simplify_tolerance (Optional[float]): All points in the simplified object will be within
+            the tolerance distance of the original geometry, in degrees.
+
+    Returns:
+        Iterator[Tuple[str, Dict[str, Any]]]: Iterator of the data extent as a geojson dict
+            for each asset.
+    """
+    asset_name_and_extent = next(
+        data_footprints_for_data_assets(
+            item,
+            asset_names=asset_names,
+            scale=scale,
+            precision=precision,
+            densification_factor=densification_factor,
+            simplify_tolerance=simplify_tolerance,
+            no_data=no_data,
+        ),
+        None,
+    )
+    if asset_name_and_extent is not None:
+        _, extent = asset_name_and_extent
+        item.geometry = extent
+        return item
+    else:
+        return None
+
+
+def data_footprints_for_data_assets(
+    item: Item,
+    *,
+    asset_names: Optional[List[str]] = None,
+    scale: int = DEFAULT_SCALE,
+    precision: int = DEFAULT_PRECISION,
+    densification_factor: Optional[int] = None,
+    simplify_tolerance: Optional[float] = None,
+    no_data: Optional[int] = None,
+) -> Iterator[Tuple[str, Dict[str, Any]]]:
+    """
+    Accepts an Item and an optional list of asset names within that item, and produces
+    an iterator over the same asset names (if they exist and have the role "data") and
```

Comment:
While I agree that `"data"` is a good default role, this should be configurable -- there could be providers that use something other than `data`.

Reply:
Ok, will add this.